src/main/java/com/example/gradecalculator/service/SubjectService.java

```diff
@@ -97,29 +101,97 @@ public TreeMap<String, Set<UserSubjectTO>> selectedSubject(long userId) {
         for (UserSubject userSubject : userSubjects) {
             String year = userSubject.getSchoolYear().getName();
             var userSubjectTO = new UserSubjectTO();
-            userSubjectTO.setID(userSubject.getId());
+            userSubjectTO.setId(userSubject.getId());
             userSubjectTO.setName(userSubject.getSubject().getName());
             subjectsByYear.computeIfAbsent(year, k -> new TreeSet<>(Comparator.comparing(UserSubjectTO::getName))).add(userSubjectTO);
         }
 
         return subjectsByYear;
     }
 
+
     @Transactional
     public boolean deleteSubject(Long subjectId, String userId) {
         Optional<UserSubject> userSubjectOpt = userSubjectRepository.findById(subjectId);
 
         if (!userSubjectOpt.isPresent()) {
```

Comment:
use isEmpty instead of checking if present and then inverting the result (reintroduced in merge?)